src/github.com/getlantern/fdcount/fdcount.go

```diff
@@ -41,16 +40,23 @@ func (c *Counter) AssertDelta(expected int) error {
 	if err != nil {
 		return err
 	}
-	actual := c.countMatches(out) - c.startingCount
+	endingLines, endingCount := c.matchingLines(out)
+	actual := endingCount - c.startingCount
 	if actual != expected {
-		return fmt.Errorf("Unexpected TCP file descriptor count. Expected %d, have %d.\n\nInitial lsof output\n-----------------------------\n%s\n\nCurrent lsof output\n-----------------------------\n%s\n",
-			expected, actual, string(c.startingOut), string(out))
+		return fmt.Errorf("Unexpected TCP file descriptor count. Expected %d, have %d.\n\n%s",
+			expected, actual, lsofDelta(c.startingLines, endingLines))
 	}
 	return nil
 }
 
-func (c *Counter) countMatches(out []byte) int {
-	return bytes.Count(out, []byte(c.match))
+func (c *Counter) matchingLines(out []byte) (string, int) {
+	lines := make([]string, 0)
```

Comment:
@oxtoacart  Is there an average or a maximum number of lines we could expect to get from this function? Do you think it would have any practical benefit if we extend the capacity of the `lines` array to prevent reallocations by `append()`?